src/setup-beam.js

```diff
@@ -338,15 +339,51 @@ function sortVersions(left, right) {
 }
 
 function getRunnerOSVersion() {
-  const ImageOSToContainer = {
+  return getContainerFromEnvImageOS() || getContainerFromOS()
+}
+
+function getContainerFromEnvImageOS() {
+  return {
     ubuntu16: 'ubuntu-16.04',
     ubuntu18: 'ubuntu-18.04',
     ubuntu20: 'ubuntu-20.04',
     win16: 'windows-2016',
     win19: 'windows-2019',
+  }[process.env.ImageOS]
+}
+
+function getContainerFromOS() {
```

Comment:
This is where the guessing game now happens. It's convoluted, sure, but I'm not sure there's a better way to fetch this info. from the target OSs

Reply:
See my comment here https://github.com/erlef/setup-beam/issues/58#issuecomment-877672764

Reply:
Per "I'd rather us not start doing guess work around operating systems" I'm reverting this guessing bit. Whatever it is, the self-hosted runner will have to use the `ImageOS` env. variable to achieve the same behaviour. I surely don't want to introduce a new env. variable and that one's already documented in the README.